package.json

```diff
@@ -27,17 +27,18 @@
     "icons:transpile": "babel ./lib/icons/react --out-dir ./lib/icons/react",
     "icons:css": "cp src/icons/style/icons.css ./lib/icons/style/icons.css",
     "icons:docs": "node ./scripts/node/icons/buildIconManifest.js",
-    "comp:jsx": "NODE_ENV=jsx_compile babel src --out-dir=lib --source-maps --no-copy-ignored",
+    "comp:jsx": "NODE_ENV=jsx_compile babel src --out-dir=lib --source-maps --no-copy-ignored --extensions '.js,.jsx,.ts,.tsx'",
     "comp:css": "postcss src/components/style/index.css -o ./lib/components/style/fds-components.css --config=./postcss.config.js",
     "comp:docs": "build-storybook -c .storybook -o ./docs/fds-components/",
     "test": "jest --color",
     "update-snapshots": "jest -u --color",
-    "lint": "eslint ./src/ --ext .js,.jsx",
+    "lint": "eslint ./src/ --ext .js,.jsx,.ts,.tsx",
     "precommit": "lint-staged",
     "storybook": "start-storybook -p 6006",
     "build-storybook": "build-storybook",
     "stats:css": "node ./scripts/node/util/getClassStats.js",
-    "stats:imports": "node ./scripts/node/util/getImportStats.js"
+    "stats:imports": "node ./scripts/node/util/getImportStats.js",
+    "type-check": "tsc --jsx react --allowSyntheticDefaultImports --noEmit"
```

Comment:
I feel like this command should be able to be simplified to `"tsc"`. It should be able to grab all of the flag values from the `tsconfig.json`.